codegen/kuberesource/crd.go

```diff
@@ -153,3 +185,85 @@ func CustomResourceDefinition(
 	}
 	return crd
 }
+
+type ErrMap map[string]error
+type CrdNeedsUpgrade struct {
+	CRDName string
+}
+
+func (e *CrdNeedsUpgrade) Error() string {
+	return fmt.Sprintf("CRD %s needs to be upgraded", e.CRDName)
+}
+
+type CrdNotFound struct {
```

Comment:
Can we simplify by just using an error with a formatted error string?

`eris.Errorf("CRD %s not found. Is it unused?", e.CRDName)`

Reply:
i wanted to a have an error type, so that the check itself can do a type cast. I treat CrdNotFound as a warning and CrdNeedsUpdate as an error, and any other error is ignored and just printed to stdsout.
see the gloo-mesh PR